### ISSUE TITLE
Add replica of Eric's Image Converter

### ISSUE THREAD

**Issue by aprzn123** [aprzn123/TheThirdCan#38]
Would be nice to be able to use it without manually messing with the javascript.

**Comment by HumanoidSandvichDispenser** [comment]
we might need an image manipulation library for this

**Comment by HumanoidSandvichDispenser** [comment]
Alternatively to an image manip library, we could use a REST API.

I'm not sure if this is too much, but I think we should also have options for additional postprocessing and different resampling methods.

**Comment by aprzn123** [comment]
That seems feasible, especially since once the backend is fully up (tsar is surprisingly hard), it'll be pretty microservice-y

**Comment by HumanoidSandvichDispenser** [comment]
I just remembered, a problem with using the REST API is that users with content filtering may not be able to use it.

**Comment by aprzn123** [comment]
True. Perhaps I can see if they can access my other domain better.

**Comment by HumanoidSandvichDispenser** [comment]
I put together something quick that can take image URLs and generate the request JSON for uploading avatars, and it can add backgrounds if the image is transparent. It was based off of a converter I made 2 years ago. Let me know if there's something I should change.

https://github.com/HumanoidSandvichDispenser/ttc-avatar-image-api

**Comment by aprzn123** [comment]
oh yeah, perhaps I should figure out how to use docker. I've been fine without it so far, but it seems useful.

**Comment by HumanoidSandvichDispenser** [comment]
Deploying the avatar image API with docker-compose is actually pretty simple.

If you have docker-compose installed, you can run `docker-compose build` to build the container and then run `docker-compose up` to deploy it.

I'll upload it soon to Docker Hub so it's easier to deploy alongside @thekifake's pronoun API.
